### ISSUE TITLE
ShortInput error while querying for A records

### ISSUE THREAD

**Issue by glts** [NLnetLabs/domain#70]
The following program (adapted from the ‘sync’ example in domain-resolv) panics:

```
use domain::{
    base::{name::Dname, Rtype},
    rdata::AllRecordData,
};
use domain_resolv::StubResolver;
use std::str::FromStr;

fn main() {
    let name = Dname::<Vec<_>>::from_str("imsavscan.netvigator.com").unwrap();

    let res = StubResolver::run(move |stub| async move {
        stub.query((name, Rtype::A)).await
    });

    let res = res.unwrap();
    let res = res.answer().unwrap().limit_to::<AllRecordData<_, _>>();
    for record in res {
        let record = record.unwrap();
        println!("{}", record);
    }
}
```

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ShortInput', src/main.rs:18:22
```

It’s just a query for A records of domain `imsavscan.netvigator.com`.

It looks like the answer contains defective records? However, other DNS resolvers that I have tried do not indicate an error with this query, so I’m wondering if this could be a bug in domain-resolv? I could not investigate further, apologies for the simple bug report.

**Comment by partim** [comment]
Curiously, it works on my machine. The response is fairly big, though. So there might be some truncation issues, perhaps caused by a NAT box along the way or some such. We do indeed currently not check for this case.

I will add a check for completeness of the message.

**Comment by partim** [comment]
In the mean time, if you need a quick workaround, you could enable the `use_vc` option for the stub resolver config. This should force TCP and avoid truncation.

**Comment by glts** [comment]
Thank you, that works. (I can reproduce this both on my machine at home and on a virtual private server in a data centre somewhere, by the way.)

**Comment by partim** [comment]
Would you perchance be able to run `dig +ignore +bufsize=1472 imsavscan.netvigator.com` on an affected machine and paste the output (perhaps cutting out the answer section if it is just a bunch of A records)?

**Comment by partim** [comment]
Scratch that. Can you try again with the _udp-buffer-size_ branch?

I seem to have used a different UDP buffer size than I actually advertised.

**Comment by glts** [comment]
On the branch I now see one more record line printed, and no panic. Great!

**Comment by partim** [comment]
This has been fixed in #71.
